packages/frontend/src/components/AccountListSidebar/AccountListSidebar.tsx

```diff
@@ -78,6 +85,91 @@ export default function AccountListSidebar({
   }
 
   const [syncAllAccounts, setSyncAllAccounts] = useState(true)
+
+  // Drag and drop handlers
+  const handleDragStart = (accountId: number) => {
+    setDraggedAccountId(accountId)
+  }
+
+  const handleDragOver = (e: React.DragEvent, index: number) => {
+    e.preventDefault()
+    if (draggedAccountId === null) return
+
+    const target = e.currentTarget as HTMLDivElement
+    const rect = target.getBoundingClientRect()
+    const position = e.clientY < rect.top + rect.height / 2 ? 'top' : 'bottom'
+
+    setDropIndicator({ index, position })
+  }
+
+  const handleDragLeave = () => {
+    setDropIndicator(null)
+  }
+
+  const handleDragEnd = () => {
+    setDraggedAccountId(null)
+    setDropIndicator(null)
+  }
+
+  const handleDrop = async (e: React.DragEvent) => {
```

Comment:
Since this function doesn't utilize `await` much, I think It'd be nice to convert it to a sync one. Sync is easier to follow.